zmq.hpp

```diff
@@ -2794,6 +2794,70 @@ inline std::ostream &operator<<(std::ostream &os, const message_t &msg)
     return os << msg.str();
 }
 
+using timer_id_t = int;
```

Comment:
This whole diff should be behind a preprocessor if, checking the min required libzmq version. Also requires C++11, but it uses `std::optional` so either require C++17 or use the `trivial_optional` for pre C++17 version (but not sure if `milliseconds` is trivial). Otherwise just require C++17.

Reply:
Maybe add `using timer_fn_t = zmq_timer_fn;`

Reply:
I think milliseconds is fine as is. But duration would be an improvement.